include/git2/notes.h

```diff
@@ -153,6 +153,33 @@ GIT_EXTERN(int) git_note_create(
 	const char *note,
 	int force);
 
+/**
+ * Add a note for an object from a commit
```

Comment:
@pks-t Adding an *optional* notes reference does seem convenient, but on the other hand, then you'd want the ability to use the current value of that reference as the parent, and that starts to make this API look complicated.